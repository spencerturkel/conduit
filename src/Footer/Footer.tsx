import * as React from 'react';
import {ComponentType, ReactNode, StatelessComponent} from 'react';

export interface LinkProps {
    children: ReactNode;
    className: string;
    to: string;
}

const Footer = (Link: ComponentType<LinkProps>): StatelessComponent => () => (
    <footer>
        <div className="container">
            <Link to="/" className="logo-font">
                conduit
            </Link>
            <span className="attribution">
                An interactive learning project from <a href="https://thinkster.io">Thinkster</a>. Code &amp; design
                licensed under MIT.
            </span>
        </div>
    </footer>
);

export default Footer;
