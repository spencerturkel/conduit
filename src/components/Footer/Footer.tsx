import * as React from "react";
import { ComponentType, StatelessComponent } from "react";

import { LinkProps } from "../link-props";
import setDisplayName from "../set-display-name";

const Footer = (Link: ComponentType<LinkProps>): StatelessComponent => {
  const sfc: StatelessComponent = () => (
    <footer>
      <div className="container">
        <Link to="/" className="logo-font">
          conduit
        </Link>
        <span className="attribution">
          An interactive learning project from{" "}
          <a href="https://thinkster.io">Thinkster</a>. Code &amp; design
          licensed under MIT.
        </span>
      </div>
    </footer>
  );

  setDisplayName(sfc, Footer, Link);

  return sfc;
};

export default Footer;
