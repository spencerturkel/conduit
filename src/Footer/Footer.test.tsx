import {mount, ReactWrapper} from 'enzyme';
import * as React from 'react';
import {ReactElement} from 'react';

import Footer, {LinkProps} from './Footer';

function Link({className, children, to}: LinkProps): ReactElement<LinkProps> {
    return (
        <a className={className} href={to}>
            {children}
        </a>
    );
}

function TestFooter(): ReactElement<{}> {
    const Result = Footer(Link);
    return <Result />;
}

let sut: ReactWrapper;

beforeEach(() => {
    sut = mount(<TestFooter />);
});

it('should match snapshot', () => {
    expect(sut).toMatchSnapshot();
});
