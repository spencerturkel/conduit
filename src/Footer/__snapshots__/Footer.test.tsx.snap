// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <TestFooter />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Object {
        "instance": <footer>
          <div
                    class="container"
          >
                    <a
                              class="logo-font"
                              href="/"
                    >
                              conduit
                    </a>
                    <span
                              class="attribution"
                    >
                              An interactive learning project from 
                              <a
                                        href="https://thinkster.io"
                              >
                                        Thinkster
                              </a>
                              . Code & design licensed under MIT.
                    </span>
          </div>
</footer>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
>
            <Link
                        className="logo-font"
                        to="/"
            >
                        conduit
            </Link>
            <span
                        className="attribution"
            >
                        An interactive learning project from 
                        <a
                                    href="https://thinkster.io"
                        >
                                    Thinkster
                        </a>
                        . Code & design licensed under MIT.
            </span>
</div>,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="container"
>
              <a
                            class="logo-font"
                            href="/"
              >
                            conduit
              </a>
              <span
                            class="attribution"
              >
                            An interactive learning project from 
                            <a
                                          href="https://thinkster.io"
                            >
                                          Thinkster
                            </a>
                            . Code & design licensed under MIT.
              </span>
</div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  className="logo-font"
                  to="/"
>
                  conduit
</Link>,
                <span
                  className="attribution"
>
                  An interactive learning project from 
                  <a
                                    href="https://thinkster.io"
                  >
                                    Thinkster
                  </a>
                  . Code & design licensed under MIT.
</span>,
              ],
              "className": "container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "conduit",
                  "className": "logo-font",
                  "to": "/",
                },
                "ref": null,
                "rendered": Object {
                  "instance": <a
                    class="logo-font"
                    href="/"
>
                    conduit
</a>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "conduit",
                    "className": "logo-font",
                    "href": "/",
                  },
                  "ref": null,
                  "rendered": Array [
                    "conduit",
                  ],
                  "type": "a",
                },
                "type": [Function],
              },
              Object {
                "instance": <span
                  class="attribution"
>
                  An interactive learning project from 
                  <a
                                    href="https://thinkster.io"
                  >
                                    Thinkster
                  </a>
                  . Code & design licensed under MIT.
</span>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "An interactive learning project from ",
                    <a
                      href="https://thinkster.io"
>
                      Thinkster
</a>,
                    ". Code & design licensed under MIT.",
                  ],
                  "className": "attribution",
                },
                "ref": null,
                "rendered": Array [
                  "An interactive learning project from ",
                  Object {
                    "instance": <a
                      href="https://thinkster.io"
>
                      Thinkster
</a>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Thinkster",
                      "href": "https://thinkster.io",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Thinkster",
                    ],
                    "type": "a",
                  },
                  ". Code & design licensed under MIT.",
                ],
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "footer",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": Object {
          "instance": <footer>
            <div
                        class="container"
            >
                        <a
                                    class="logo-font"
                                    href="/"
                        >
                                    conduit
                        </a>
                        <span
                                    class="attribution"
                        >
                                    An interactive learning project from 
                                    <a
                                                href="https://thinkster.io"
                                    >
                                                Thinkster
                                    </a>
                                    . Code & design licensed under MIT.
                        </span>
            </div>
</footer>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
>
              <Link
                            className="logo-font"
                            to="/"
              >
                            conduit
              </Link>
              <span
                            className="attribution"
              >
                            An interactive learning project from 
                            <a
                                          href="https://thinkster.io"
                            >
                                          Thinkster
                            </a>
                            . Code & design licensed under MIT.
              </span>
</div>,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <div
                class="container"
>
                <a
                                class="logo-font"
                                href="/"
                >
                                conduit
                </a>
                <span
                                class="attribution"
                >
                                An interactive learning project from 
                                <a
                                                href="https://thinkster.io"
                                >
                                                Thinkster
                                </a>
                                . Code & design licensed under MIT.
                </span>
</div>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    className="logo-font"
                    to="/"
>
                    conduit
</Link>,
                  <span
                    className="attribution"
>
                    An interactive learning project from 
                    <a
                                        href="https://thinkster.io"
                    >
                                        Thinkster
                    </a>
                    . Code & design licensed under MIT.
</span>,
                ],
                "className": "container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "conduit",
                    "className": "logo-font",
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": <a
                      class="logo-font"
                      href="/"
>
                      conduit
</a>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "conduit",
                      "className": "logo-font",
                      "href": "/",
                    },
                    "ref": null,
                    "rendered": Array [
                      "conduit",
                    ],
                    "type": "a",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": <span
                    class="attribution"
>
                    An interactive learning project from 
                    <a
                                        href="https://thinkster.io"
                    >
                                        Thinkster
                    </a>
                    . Code & design licensed under MIT.
</span>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "An interactive learning project from ",
                      <a
                        href="https://thinkster.io"
>
                        Thinkster
</a>,
                      ". Code & design licensed under MIT.",
                    ],
                    "className": "attribution",
                  },
                  "ref": null,
                  "rendered": Array [
                    "An interactive learning project from ",
                    Object {
                      "instance": <a
                        href="https://thinkster.io"
>
                        Thinkster
</a>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Thinkster",
                        "href": "https://thinkster.io",
                      },
                      "ref": null,
                      "rendered": Array [
                        "Thinkster",
                      ],
                      "type": "a",
                    },
                    ". Code & design licensed under MIT.",
                  ],
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": "footer",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
